default/python/AI/Trait/CharacterStrings.py

```diff
@@ -0,0 +1,51 @@
+"""
+CharacterStrings.py contains tables of strings indexed by the AI character
+"""
+
+import Trait.Character
+import freeOrionAIInterface as fo  # pylint: disable=import-error
+
+
+class CharacterTable(object):
+    """A table indexed by a particular trait of a Character."""
+
+    def __init__(self, trait_class, table, post_process_func=None):
+        """Store trait type, table and post processing function."""
+        self.trait_class = trait_class
+        self.table = table
+        if None not in self.table:
+            table[None] = "UNKNOWN_VALUE_SYMBOL"
+        self.post_process = post_process_func
+
+    def __repr__(self):
+        return str(self.table)
+
+    def __len__(self):
+        return len(self.table)
+
+    def __getitem__(self, character):
+        """Return the indexed and post-processed string. Get the key from the correct trait in the character."""
+        elem = self.table[character.get_trait(self.trait_class).key]
+        if self.post_process:
+            elem = self.post_process(elem)
+        print "CharacterTable returns ", elem
+        return elem
+
+
+_aggression_label_suffix = {fo.aggression.beginner: "_BEGINNER",
+                            fo.aggression.turtle: "_TURTLE",
+                            fo.aggression.cautious: "_CAUTIOUS",
+                            fo.aggression.typical: "_TYPICAL",
+                            fo.aggression.aggressive: "_AGGRESSIVE",
+                            fo.aggression.maniacal: "_MANIACAL"}
+
+
+def make_aggression_table(prefix, post_process_func=None):
```

Comment:
I changed the function name to  `get_trait_name_aggression(character)` .  I made functions not used outside of `character_strings_module.py`, except in tests, private.